testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -5909,6 +5913,227 @@ private void testMaterializedViewColumnName(String columnName, boolean delimited
         assertUpdate("DROP MATERIALIZED VIEW " + viewName);
     }
 
+    @Test
+    public void testProjectionPushdown()
+    {
+        if (hasBehavior(SUPPORTS_CREATE_TABLE_WITH_DATA) && hasBehavior(SUPPORTS_ROW_TYPE)) {
+            try (TestTable testTable = new TestTable(
+                    getQueryRunner()::execute,
+                    "test_projection_pushdown_",
+                    "(id BIGINT, root ROW(f1 BIGINT, f2 BIGINT))",
+                    ImmutableList.of("(1, ROW(1, 2))", "(2, NULl)", "(3, ROW(NULL, 4))"))) {
+                String selectQuery = "SELECT root.f1, id FROM " + testTable.getName();
+                String expectedResult = "VALUES (BIGINT '1', BIGINT '1'), (NULL, BIGINT '2'), (NULL, BIGINT '3')";
```

Comment:
nit: Can we change the root.f1 & id columns in SELECT query? I didn't understand why these values are expected few seconds because the column order is different from the table definition.